nshlib/nsh_syscmds.c

```diff
@@ -780,8 +781,12 @@ int cmd_uname(FAR struct nsh_vtbl_s *vtbl, int argc, FAR char **argv)
   /* Process each option */
 
   first = true;
-  lib_memoutstream(&stream, alloca(sizeof(struct utsname)),
-                   sizeof(struct utsname));
+#ifdef __GNUC__
+  putsname = alloca(sizeof(struct utsname));
```

Comment:
let's declare "struct utsname xxx" directly

Reply:
ustname structure is quite large, how about maintain the current style to avoid stack overflow?

Reply:
Done